src/main.rs

```diff
@@ -93,3 +101,64 @@ fn main() {
         thread::sleep(time::Duration::from_millis(SLEEP_TIME_MS));
     }
 }
+
+fn change_gain(increase: bool) {
+    // Retrieve current gain
+    let current_gain = get_gain();
+    let mut new_gain = current_gain;
+    if increase {
+        new_gain += GAIN_OFFSET;
+    } else {
+        new_gain -= GAIN_OFFSET;
+    }
+    // Clamp to make sure the gain is in the correct range
+    let clamped_gain = clamp(new_gain, MINIDSP_GAIN_MIN, MINIDSP_GAIN_MAX);
+    if gain_changed(current_gain, clamped_gain) {
+        apply_gain(&format!("{:.1}", clamped_gain));
+    }
+}
+
+fn act_on_event(event: evdev_rs::InputEvent) {
+    const VOLUME_UP_KEY: evdev_rs::enums::EV_KEY = evdev_rs::enums::EV_KEY::KEY_U;
```

Comment:
as simple as _what the fuck_ am I reading <3 

Looks very complicated but I understand it hahaha